dockerfiles/vendor.Dockerfile

```diff
@@ -1,6 +1,6 @@
 # syntax=docker/dockerfile:1
 
-ARG GO_VERSION=1.23.2
+ARG GO_VERSION=1.22.7
```

Comment:
I am wondering if sticking to `1.23.2` across all `Dockerfile`s makes things more maintainable...I'm leaning towards that rather than having this specific `Dockerfile` Go version be set to a different version than the other ones. I get that means having the `toolchain` stanza in the `go.mod` but I think that'll probably be the most "sane" solution for the maintainers 🤔  any thoughts on this @Jamstah @thaJeztah  ?

Reply:
Yeah, not sure what the solution is to that. It's... weird. Basically says; "you can build this code with go1.22, but you must use the go1.23 version of go1.22 to use it?". The patch versions make even less sense, but I guess those are forces upon us by upstreams 😞